pkg/backends/datadog/datadog.go

```diff
@@ -203,6 +204,20 @@ func (d *Client) processMetrics(metrics *gostatsd.MetricMap, cb func(*timeSeries
 		fl.maybeFlush()
 	})
 
+	metrics.Timers.Each(func(key, tagsKey string, timer gostatsd.Timer) {
+		if timer.Histogram != nil {
+			for histogramThreshold, count := range timer.Histogram {
+				bucketTag := "le:+Inf"
+				if !math.IsInf(histogramThreshold.Le, 1) {
+					bucketTag = "le:" + strconv.FormatFloat(histogramThreshold.Le, 'f', -1, 64)
+				}
+				newTags := timer.Tags.Concat([]string{bucketTag})
+				fl.addMetricf(counter, float64(count), timer.Hostname, newTags, "%s.histogram", key)
```

Comment:
I think I don't understand all the implications of introducing `counter` to `datadog` backend. It was implemented like that in #222. What's the benefit of having that? I don't really want to make a risky change if it's not really needed.

Reply:
The hazard is there's a time component to both a rate and a count, but not a gauge.

It's the number of things that occurred in (for example) 10 seconds.  If you miss a flush interval, then you collect 20 seconds of data, but you're telling upstream it's 10 seconds worth.  If you know the window you can go from counter to rate (or vice versa), but once the data leaves gostatsd, you don't know the window anymore.

It's not directly a risk (because it's a new thing), but rather things may not behave as expected when querying.  You can't test it in gostatsd, because it's how the backend handles the query.  Then if it turns out you then need to change it, **that's** a breaking change.

This is why I'm advising to test it end to end, with the real pipeline, with both happy and sad conditions :)